### PR TITLE
fix UHDAtomics.cmake to read custom Boost install prefix

### LINKED ISSUE #632: Failed 4.3.0.0 Source Build with Custom Prefix-Installed Boost Due To CMake Issue
## Issue Description ##
The newest 4.3.0.0 release fails to build from source when attempting to process UHDAtomics.cmake.

## Setup Details ##
UHD 4.3.0.0 (Downloaded directly from Releases tar.gz)
Xubuntu (Ubuntu 20.04.4 LTS)
Boost 1.80.0, compiled from source using ./bootstrap.sh --prefix=/my/custom/path

## Expected Behavior ##
Build should complete.

## Actual Behaviour ##
Build fails here:

> Configuring atomics support...
> Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB
> Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB - Success
> Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
> Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
> BOOST ATOMICS COMPILE INCLUDE: /home/seoxubuntu/boost_1_80_0-install/include
> CMAKE_REQUIRED_INCLUDES: 
> Performing Test HAVE_CXX_BOOST_ATOMICS_WITHOUT_LIB
> Performing Test HAVE_CXX_BOOST_ATOMICS_WITHOUT_LIB - Failed
> Looking for __atomic_fetch_add_4 in atomic
> Looking for __atomic_fetch_add_4 in atomic - found
> BOOST ATOMICS COMPILE INCLUDE: /home/seoxubuntu/boost_1_80_0-install/include
> CMAKE_REQUIRED_INCLUDES: 
> Performing Test HAVE_CXX_BOOST_ATOMICS_WITH_LIB
> Performing Test HAVE_CXX_BOOST_ATOMICS_WITH_LIB - Failed
> CMake Error at cmake/Modules/UHDAtomics.cmake:105 (message):
>   Host compiler must support std::atomic!
> Call Stack (most recent call first):
>   lib/utils/CMakeLists.txt:187 (CHECK_ATOMICS_LIB_REQUIRED)
>   lib/CMakeLists.txt:46 (include)
>   lib/CMakeLists.txt:93 (INCLUDE_SUBDIRECTORY)

## Steps to reproduce the problem ##
Download Boost 1.80.0, run bootstrap.sh with custom prefix.
Add prefix to Boost_INCLUDE_DIR during cmake of UHD 4.3.0.0.
Observe error.

## Additional Information ##
It appears that according to [here](https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html) the `try_compile()` used by CMake does not acquire the included directories. Hence the fix is to simply set the environment variable `CMAKE_REQUIRED_INCLUDES` appropriately. I have done this internally, and it appears to fix the problem.

### PR DESCRIPTION
**Pull request by icyveins7**
# Pull Request Details

## Description

This fixes a CMake build issue with custom prefix-installed Boost libraries.

## Related Issue

Resolves #632

## Which devices/areas does this affect?
NA, this is a build issue.

## Testing Done
CMake proceeds.

## Checklist

- [ x] I have read the CONTRIBUTING document.
- [ x] My code follows the code style of this project. See CODING.md.
- [ x] I have updated the documentation accordingly.
- [ x] I have added tests to cover my changes, and all previous tests pass.
- [ x] I have checked all compat numbers if they need updating (FPGA compat,
      MPM compat, noc_shell, specific RFNoC block, ...)

### CODE REVIEW

**Comment by icyveins7** [comment]
_I have read the CLA Document and I hereby sign the CLA_

**Comment by icyveins7** [comment]
recheck

**Comment by chockey** [comment]
For what it's worth, I have the same compile problem and had applied this exact same fix to my local tree to fix it.

File: host/cmake/Modules/UHDAtomics.cmake [hunk 1]
```diff
@@ -36,6 +36,7 @@ endfunction(CHECK_WORKING_CXX_ATOMICS64)
 # Note: If we reach this, we have already checked for the existence of Boost,
 # and Boost.Lockfree is a header-only library so no linker options required.
 function(CHECK_WORKING_CXX_BOOST_ATOMICS varname)
+    set(CMAKE_REQUIRED_INCLUDES ${Boost_INCLUDE_DIRS})
```
Comment (mbr0wn) [inline host/cmake/Modules/UHDAtomics.cmake:39]: We need to `unset()` this after the check; but otherwise this is good.

**Comment by mbr0wn** [comment]
I'm putting this into our internal review/CI queue.

**Comment by mbr0wn** [comment]
Thanks for this fix!

**Pull request closed without merging**